File: makeitone.py
def makeOne(inputnum, cntBase):
    cnt = [cntBase, cntBase, cntBase, cntBase, cntBase]
    minCnt = INT_MAX
    if(inputnum > 1):
        cnt[0] = ((inputnum % 3 == 0) and (makeOne(inputnum/3, cnt[0] + 1)) or INT_MAX)
        cnt[1] = ((inputnum % 2 == 0) and (makeOne(inputnum/2, cnt[1] + 1)) or INT_MAX)
        cnt[2] = (((inputnum - 1) % 3 == 0) and (makeOne((inputnum - 1)/3, cnt[2] + 2)) or INT_MAX)
        cnt[3] = (((inputnum - 2) % 3 == 0) and (makeOne((inputnum - 2)/3, cnt[3] + 3)) or INT_MAX)
        cnt[4] = (((inputnum - 1) % 2 == 0) and (makeOne((inputnum - 1)/2, cnt[4] + 2)) or INT_MAX)
    
    if(cnt[0] < minCnt):
        minCnt = cnt[0]
    if(cnt[1] < minCnt):
        minCnt = cnt[1]
    if(cnt[2] < minCnt):
        minCnt = cnt[2]
    if(cnt[3] < minCnt):
        minCnt = cnt[3]
    if(cnt[4] < minCnt):
        minCnt = cnt[4]    
    return minCnt

cntBase = 0
inputnum = input()   
INT_MAX = 99999999
print(makeOne(inputnum, cntBase))

File: makeitone.c
#include <stdio.h>

#define INT_MAX 99999999

int makeOne(int, int);

int main(void){
    int input = 0;
    int cnt = 0;
    
    scanf("%d", &input);
    printf("%d", makeOne(input, cnt));
    
    return 0;
}

int makeOne(int input, int cntBase){
    int cnt[5];
    int minCnt = INT_MAX;
    int i = 0;
    
    for (i = 0; i < 5; i++){
    	cnt[i] = cntBase;
	}

	    if(input > 1){
	        if(input % 3 == 0){
	            cnt[0] = makeOne(input/3, cnt[0] + 1);
	        }
	        else{
	        	cnt[0] = INT_MAX;
			}
	
	        if(input % 2 == 0){
	            cnt[1] = makeOne(input/2, cnt[1] + 1);
	        }      
	        else{
	        	cnt[1] = INT_MAX;
			}
	        
	        if((input - 1) % 3 == 0){
	            cnt[2] = makeOne((input - 1)/3, cnt[2] + 2);
	        }
	        else{
	        	cnt[2] = INT_MAX;
			}
			
			if((input - 2) % 3 == 0){
				cnt[3] = makeOne((input - 2)/3, cnt[3] + 3);
			}
			else{
				cnt[3] = INT_MAX;
			}
			
			if((input - 1) % 2 == 0){
				cnt[4] = makeOne((input - 1)/2, cnt[4] + 2);
			}
			else{
				cnt[4] = INT_MAX;
			}
	    }
	    
    if(cnt[0] < minCnt){
        minCnt = cnt[0];
    }
    if(cnt[1] < minCnt){
        minCnt = cnt[1];
    }
    if(cnt[2] < minCnt){
        minCnt = cnt[2];
    }
    if(cnt[3] < minCnt){
    	minCnt = cnt[3];
	}
	if(cnt[4] < minCnt){
		minCnt = cnt[4];
	}
	
    return minCnt;
}
